### PR TITLE
Add a note to the doc of Evaluator.evaluate

### PR DESCRIPTION
**Pull request by mitmul**
In Chainer, "no backprop" and "evaluation" are orthogonal concept, but chainer.training.extensions.Evaluator.evaluate calls `chainer.no_backprop_mode()` before executing `eval_func`. It should be explicitly explained in the document.

### CODE REVIEW

File: chainer/training/extensions/evaluator.py [hunk 1]
```diff
@@ -148,6 +148,14 @@ def evaluate(self):
 
         Users can override this method to customize the evaluation routine.
 
+        .. note::
+
+            This method enclose :meth:`eval_func` calls in
```
Comment (delta2323) [inline chainer/training/extensions/evaluator.py]: in -> with

**Comment by mitmul** [comment]
@delta2323 Thanks for the comment. I fiexed it.

**Comment by delta2323** [comment]
jenkins, test this please

**Comment by mitmul** [comment]
Jenkins, test this please

File: chainer/training/extensions/evaluator.py [hunk 2]
```diff
@@ -148,6 +148,14 @@ def evaluate(self):
 
         Users can override this method to customize the evaluation routine.
 
+        .. note::
+
+            This method encloses :meth:`eval_func` calls with
+            `with function.no_backprop_mode()` context. So, all calculations
```
Comment (niboshi) [inline chainer/training/extensions/evaluator.py]: It's not linked. How about
``` `with function.no_backprop_mode()` context.```
->
```:func:`chainer.no_backprop_mode` context.```

File: chainer/training/extensions/evaluator.py [hunk 3]
```diff
@@ -148,6 +148,14 @@ def evaluate(self):
 
         Users can override this method to customize the evaluation routine.
 
+        .. note::
+
+            This method encloses :meth:`eval_func` calls with
```
Comment (niboshi) [inline chainer/training/extensions/evaluator.py]: `eval_func` is not a method. How about ```:attr:``` ?

File: chainer/training/extensions/evaluator.py [hunk 4]
```diff
@@ -148,6 +148,14 @@ def evaluate(self):
...
+            `with function.no_backprop_mode()` context. So, all calculations
+            using :class:`~chainer.Function`\ s inside :meth:`eval_func` don't
```
Comment (niboshi) [inline chainer/training/extensions/evaluator.py]: Ditto. (```:attr:```)

**Comment by mitmul** [comment]
@niboshi Thanks, I resolved them.

@niboshi @delta2323 Could you approve this PR?

File: chainer/training/extensions/evaluator.py [hunk 5]
```diff
@@ -148,6 +148,14 @@ def evaluate(self):
 
         Users can override this method to customize the evaluation routine.
 
+        .. note::
+
+            This method encloses :attr:`eval_func` calls with
+            :func:`function.no_backprop_mode` context. So, all calculations
```
Comment (niboshi) [inline chainer/training/extensions/evaluator.py]: It would be better:
... context, so all calculations...

**Comment by delta2323** [comment]
LGTM

**Comment by mitmul** [comment]
@delta2323 Could you merge this?

**Comment by mitmul** [comment]
@niboshi @delta2323 I reflected your suggestions. Could you check it again and merge this PR?

**Pull request merged**